### PR TITLE
Make to_shmem Gecko-only

### PR DESCRIPTION
**Pull request by SimonSapin**
As suggested in https://github.com/servo/servo/issues/26713#issuecomment-638185219

### CODE REVIEW

File: components/style/gecko_string_cache/namespace.rs [hunk 1]
```diff
@@ -25,17 +25,9 @@ macro_rules! ns {
 
 /// A Gecko namespace is just a wrapped atom.
 #[derive(
-    Clone,
```
Comment (jdm) [inline components/style/gecko_string_cache/namespace.rs:28]: I don't think we need to add feature checks to any of the style/gecko/ or style/gecko_string_cache files.

**Review by jdm — APPROVED** [review]

File: components/style/gecko_string_cache/namespace.rs (hunk 1 above)
Comment (SimonSapin) [reply components/style/gecko_string_cache/namespace.rs]: Good point. Amended.

**Comment by emilio** [comment]
@bors-servo r-

Can we discuss this for a second? This is going to make syncs even more painful and if I understand correctly this affects llvm IR but not generated code, right?

**Comment by SimonSapin** [comment]
Correct, this is about reducing compile times. I would expect the linker to remove functions that are never called. (Maybe even without LTO? IIRC with `--gc-sections` is the default in rustc.)

I hadn’t considered the indirect impact on sync as you explained it on Matrix (mozilla-central contributors accidentally breaking the Servo configuration more often as it diverges more).

`to_shmem_slice_ptr` does show up in https://github.com/servo/servo/issues/26713#issuecomment-638112283 but with only 0.4% of IR lines in libscript. This is probably not worth the downside, so closing.

**Pull request closed without merging**